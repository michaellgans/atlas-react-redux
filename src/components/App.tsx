function App() {
  return <>Hello Atlas</>;
}

export default App;
