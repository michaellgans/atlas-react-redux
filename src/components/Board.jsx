// Board Component

// Asset Imports

// Script Imports

// Returns a Board Component
export function Board() {
    // Define Hook
    return (
    );
}
