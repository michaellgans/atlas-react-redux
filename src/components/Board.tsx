// Board Component

// Asset Imports
import { useAppSelector } from "../store";
import { List } from "./List";
import { RootState } from "../store";

// Returns a Board Component
export function Board() {
  // Define Hook
  const lists = useAppSelector((state: RootState) => state.listsSlice.items);

  return (
    // START BOARD
    <div className="m-auto h-screen w-screen overflow-x-scroll text-center">
      <div className="flex h-full space-x-4">
        {lists.map((list) => (
          <List key={list.id} listID={list.id} listTitle={list.title} />
        ))}
      </div>
    </div>
  );
}
