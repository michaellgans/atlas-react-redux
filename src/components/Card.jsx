// Card Component

// Asset Imports

// Script Imports

// Returns a Card Component
export function Card() {
    // Define Hook
    return (
    );
}
