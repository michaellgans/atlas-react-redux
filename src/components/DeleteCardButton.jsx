// DeleteCardButton Component

// Asset Imports

// Script Imports

// Returns a DeleteCardButton Component
export function DeleteCardButton() {
    // Define Hook
    return (
    );
}
