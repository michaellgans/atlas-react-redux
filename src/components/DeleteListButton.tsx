// DeleteListButton Component

// Asset Imports
import { deleteList } from "../slices/listsSlice";
import { useAppDispatch } from "../store";

// Types
type DeleteListButtonProps = {
  listID: string;
};

// Returns a DeleteListButton Component
export function DeleteListButton({ listID }: DeleteListButtonProps) {
  // Define Hook
  const dispatch = useAppDispatch();

  const handleDeleteList = () => {
    dispatch(deleteList(listID));
  };

  return (
    <button className="h-[30px]" onClick={handleDeleteList}>
      <svg
        className="hidden h-[30px] w-[30px] cursor-pointer group-hover/list:block"
        xmlns="http://www.w3.org/2000/svg"
        fill="none"
        viewBox="0 0 24 24"
        strokeWidth="{1.5}"
        stroke="currentColor"
      >
        <path
          strokeLinecap="round"
          strokeLinejoin="round"
          d="m9.75 9.75 4.5 4.5m0-4.5-4.5 4.5M21 12a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
        />
      </svg>
    </button>
  );
}
