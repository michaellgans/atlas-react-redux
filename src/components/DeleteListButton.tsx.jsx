// DeleteListButton.tsx Component

// Asset Imports

// Script Imports

// Returns a DeleteListButton.tsx Component
export function DeleteListButton.tsx() {
    // Define Hook
    return (
    );
}
