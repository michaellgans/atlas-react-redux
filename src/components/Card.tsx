// Card Component

// Asset Imports
import { DeleteCardButton } from "./DeleteCardButton";

// Script Imports

// Types
type CardProps = {
  title: string;
  description: string;
};

// Returns a Card Component
export function Card({ title, description }: CardProps) {
  // Define Hook
  return (
    <div className="card group/card m-3 flex min-h-24 w-full flex-col items-start rounded bg-off-white-light px-4 py-2 text-blue">
      <h5 className="my-2 flex w-full items-end justify-between text-xl font-black">
        <span>{title}</span>
        {/* DELETE CARD BUTTON */}
        <DeleteCardButton />
      </h5>
      <p className="mt-0 text-left">{description}</p>
    </div>
  );
}
