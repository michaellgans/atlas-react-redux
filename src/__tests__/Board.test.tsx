// Test for Board Component

// Imports
import { render } from "@testing-library/react";
import { expect, test, it } from 'vitest';
import { Board } from "../components/Board";

it('Controll Test', () => {
    expect(2 + 2).toBe(4);
});

// test('Board renders correctly', () => {
//     const { container } = render(<Board />);
//     expect(container).toMatchSnapshot();
// });
