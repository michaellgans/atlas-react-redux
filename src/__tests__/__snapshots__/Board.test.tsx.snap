// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Board renders correctly 1`] = `
<div>
  <div
    class="m-auto h-screen w-screen overflow-x-scroll text-center"
  >
    <div
      class="flex h-full space-x-4"
    >
      <div
        class="group/list h-full min-w-96 p-4"
      >
        <button
          class="h-[30px]"
        >
          <svg
            class="hidden h-[30px] w-[30px] cursor-pointer group-hover/list:block"
            fill="none"
            stroke="currentColor"
            stroke-width="{1.5}"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m9.75 9.75 4.5 4.5m0-4.5-4.5 4.5M21 12a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </button>
        <h3>
          List 1
        </h3>
        <div
          class="card group/card m-3 flex min-h-24 w-full flex-col items-start rounded bg-off-white-light px-4 py-2 text-blue"
        >
          <h5
            class="my-2 flex w-full items-end justify-between text-xl font-black"
          >
            <span>
              Title
            </span>
            <button
              class="hidden group-hover/card:block"
            >
              <svg
                class="h-[20px] w-[20px]"
                fill="none"
                stroke="#00003c"
                stroke-width="{1.5}"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m14.74 9-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 0 1-2.244 2.077H8.084a2.25 2.25 0 0 1-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 0 0-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 0 1 3.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 0 0-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 0 0-7.5 0"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </button>
          </h5>
          <p
            class="mt-0 text-left"
          >
            test...
          </p>
        </div>
        <div
          class="card group/card m-3 flex min-h-24 w-full flex-col items-start rounded bg-off-white-light px-4 py-2 text-blue"
        >
          <h5
            class="my-2 flex w-full items-end justify-between text-xl font-black"
          >
            <span>
              Title
            </span>
            <button
              class="hidden group-hover/card:block"
            >
              <svg
                class="h-[20px] w-[20px]"
                fill="none"
                stroke="#00003c"
                stroke-width="{1.5}"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m14.74 9-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 0 1-2.244 2.077H8.084a2.25 2.25 0 0 1-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 0 0-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 0 1 3.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 0 0-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 0 0-7.5 0"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </button>
          </h5>
          <p
            class="mt-0 text-left"
          >
            test...
          </p>
        </div>
        <div
          class="card group/card m-3 flex min-h-24 w-full flex-col items-start rounded bg-off-white-light px-4 py-2 text-blue"
        >
          <h5
            class="my-2 flex w-full items-end justify-between text-xl font-black"
          >
            <span>
              Title
            </span>
            <button
              class="hidden group-hover/card:block"
            >
              <svg
                class="h-[20px] w-[20px]"
                fill="none"
                stroke="#00003c"
                stroke-width="{1.5}"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m14.74 9-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 0 1-2.244 2.077H8.084a2.25 2.25 0 0 1-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 0 0-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 0 1 3.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 0 0-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 0 0-7.5 0"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </button>
          </h5>
          <p
            class="mt-0 text-left"
          >
            test...
          </p>
        </div>
        <div
          class="group/new-card m-3 flex h-44 w-full justify-center"
        >
          <form
            class="hidden min-h-24 w-full flex-col items-start rounded bg-off-white-light px-4 text-blue group-hover/new-card:flex"
          >
            <input
              class="w-11/12 resize-none overflow-auto rounded-t-3xl border-0 bg-off-white-light px-0 py-6 text-xl font-black text-blue outline-none"
              name="title"
              placeholder="Title"
              type="text"
            />
            <textarea
              class="w-11/12 resize-none overflow-auto border-0 bg-off-white-light text-blue outline-none"
              name="description"
              placeholder="Description"
            />
            <div
              class="buttons w-full"
            >
              <button
                class="w-full p-4"
                type="submit"
              >
                Save
              </button>
            </div>
          </form>
        </div>
      </div>
      <div
        class="group/list h-full min-w-96 p-4"
      >
        <button
          class="h-[30px]"
        >
          <svg
            class="hidden h-[30px] w-[30px] cursor-pointer group-hover/list:block"
            fill="none"
            stroke="currentColor"
            stroke-width="{1.5}"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m9.75 9.75 4.5 4.5m0-4.5-4.5 4.5M21 12a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </button>
        <h3>
          List 2
        </h3>
        <div
          class="card group/card m-3 flex min-h-24 w-full flex-col items-start rounded bg-off-white-light px-4 py-2 text-blue"
        >
          <h5
            class="my-2 flex w-full items-end justify-between text-xl font-black"
          >
            <span>
              Title
            </span>
            <button
              class="hidden group-hover/card:block"
            >
              <svg
                class="h-[20px] w-[20px]"
                fill="none"
                stroke="#00003c"
                stroke-width="{1.5}"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m14.74 9-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 0 1-2.244 2.077H8.084a2.25 2.25 0 0 1-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 0 0-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 0 1 3.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 0 0-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 0 0-7.5 0"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </button>
          </h5>
          <p
            class="mt-0 text-left"
          >
            test...
          </p>
        </div>
        <div
          class="card group/card m-3 flex min-h-24 w-full flex-col items-start rounded bg-off-white-light px-4 py-2 text-blue"
        >
          <h5
            class="my-2 flex w-full items-end justify-between text-xl font-black"
          >
            <span>
              Title
            </span>
            <button
              class="hidden group-hover/card:block"
            >
              <svg
                class="h-[20px] w-[20px]"
                fill="none"
                stroke="#00003c"
                stroke-width="{1.5}"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m14.74 9-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 0 1-2.244 2.077H8.084a2.25 2.25 0 0 1-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 0 0-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 0 1 3.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 0 0-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 0 0-7.5 0"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </button>
          </h5>
          <p
            class="mt-0 text-left"
          >
            test...
          </p>
        </div>
        <div
          class="card group/card m-3 flex min-h-24 w-full flex-col items-start rounded bg-off-white-light px-4 py-2 text-blue"
        >
          <h5
            class="my-2 flex w-full items-end justify-between text-xl font-black"
          >
            <span>
              Title
            </span>
            <button
              class="hidden group-hover/card:block"
            >
              <svg
                class="h-[20px] w-[20px]"
                fill="none"
                stroke="#00003c"
                stroke-width="{1.5}"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m14.74 9-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 0 1-2.244 2.077H8.084a2.25 2.25 0 0 1-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 0 0-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 0 1 3.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 0 0-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 0 0-7.5 0"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </button>
          </h5>
          <p
            class="mt-0 text-left"
          >
            test...
          </p>
        </div>
        <div
          class="group/new-card m-3 flex h-44 w-full justify-center"
        >
          <form
            class="hidden min-h-24 w-full flex-col items-start rounded bg-off-white-light px-4 text-blue group-hover/new-card:flex"
          >
            <input
              class="w-11/12 resize-none overflow-auto rounded-t-3xl border-0 bg-off-white-light px-0 py-6 text-xl font-black text-blue outline-none"
              name="title"
              placeholder="Title"
              type="text"
            />
            <textarea
              class="w-11/12 resize-none overflow-auto border-0 bg-off-white-light text-blue outline-none"
              name="description"
              placeholder="Description"
            />
            <div
              class="buttons w-full"
            >
              <button
                class="w-full p-4"
                type="submit"
              >
                Save
              </button>
            </div>
          </form>
        </div>
      </div>
      <div
        class="group/list h-full min-w-96 p-4"
      >
        <button
          class="h-[30px]"
        >
          <svg
            class="hidden h-[30px] w-[30px] cursor-pointer group-hover/list:block"
            fill="none"
            stroke="currentColor"
            stroke-width="{1.5}"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m9.75 9.75 4.5 4.5m0-4.5-4.5 4.5M21 12a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </button>
        <h3>
          List 3
        </h3>
        <div
          class="card group/card m-3 flex min-h-24 w-full flex-col items-start rounded bg-off-white-light px-4 py-2 text-blue"
        >
          <h5
            class="my-2 flex w-full items-end justify-between text-xl font-black"
          >
            <span>
              Title
            </span>
            <button
              class="hidden group-hover/card:block"
            >
              <svg
                class="h-[20px] w-[20px]"
                fill="none"
                stroke="#00003c"
                stroke-width="{1.5}"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m14.74 9-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 0 1-2.244 2.077H8.084a2.25 2.25 0 0 1-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 0 0-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 0 1 3.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 0 0-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 0 0-7.5 0"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </button>
          </h5>
          <p
            class="mt-0 text-left"
          >
            test...
          </p>
        </div>
        <div
          class="card group/card m-3 flex min-h-24 w-full flex-col items-start rounded bg-off-white-light px-4 py-2 text-blue"
        >
          <h5
            class="my-2 flex w-full items-end justify-between text-xl font-black"
          >
            <span>
              Title
            </span>
            <button
              class="hidden group-hover/card:block"
            >
              <svg
                class="h-[20px] w-[20px]"
                fill="none"
                stroke="#00003c"
                stroke-width="{1.5}"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m14.74 9-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 0 1-2.244 2.077H8.084a2.25 2.25 0 0 1-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 0 0-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 0 1 3.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 0 0-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 0 0-7.5 0"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </button>
          </h5>
          <p
            class="mt-0 text-left"
          >
            test...
          </p>
        </div>
        <div
          class="card group/card m-3 flex min-h-24 w-full flex-col items-start rounded bg-off-white-light px-4 py-2 text-blue"
        >
          <h5
            class="my-2 flex w-full items-end justify-between text-xl font-black"
          >
            <span>
              Title
            </span>
            <button
              class="hidden group-hover/card:block"
            >
              <svg
                class="h-[20px] w-[20px]"
                fill="none"
                stroke="#00003c"
                stroke-width="{1.5}"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m14.74 9-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 0 1-2.244 2.077H8.084a2.25 2.25 0 0 1-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 0 0-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 0 1 3.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 0 0-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 0 0-7.5 0"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </button>
          </h5>
          <p
            class="mt-0 text-left"
          >
            test...
          </p>
        </div>
        <div
          class="group/new-card m-3 flex h-44 w-full justify-center"
        >
          <form
            class="hidden min-h-24 w-full flex-col items-start rounded bg-off-white-light px-4 text-blue group-hover/new-card:flex"
          >
            <input
              class="w-11/12 resize-none overflow-auto rounded-t-3xl border-0 bg-off-white-light px-0 py-6 text-xl font-black text-blue outline-none"
              name="title"
              placeholder="Title"
              type="text"
            />
            <textarea
              class="w-11/12 resize-none overflow-auto border-0 bg-off-white-light text-blue outline-none"
              name="description"
              placeholder="Description"
            />
            <div
              class="buttons w-full"
            >
              <button
                class="w-full p-4"
                type="submit"
              >
                Save
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
